src/hexer/duplifier.nim

```diff
@@ -786,9 +786,22 @@ proc trDeref(c: var Context; n: var Cursor) =
     c.dest.addIntLit(0, info) # inheritance
   takeParRi c.dest, n
 
+proc trTypeDecl(c: var Context; n: Cursor) =
+  var n = n
+  var declStart = n
+  inc n
+  let symId = n.symId
+  if isLocalDecl(symId):
+    var dest = createTokenBuf()
+    takeTree(dest, declStart)
+    publish(symId, dest)
+
 proc tr(c: var Context; n: var Cursor; e: Expects) =
   if n.kind == Symbol:
     trLocation c, n, e
+  elif n.stmtKind == TypeS:
+    trTypeDecl c, n
+    takeTree c.dest, n
```

Comment:
I don't understand. So you copy the type declaration twice?!

Reply:
`trTypeDecl` is used to publish the local type decl in the `mem` so that `getType` will work. I improved the code